src/component/panels/SummaryPanel/CorrelationTable/Utilities.js

```diff
@@ -0,0 +1,275 @@
+          [],
+        ).length > 0;
+      return checkAtomType === true
+        ? getAtomType(_experiment.info.nucleus) === experimentKey && hasValues
+        : hasValues;
+    });
+
+  if (_experiments.length > 0) {
+    experimentsType[experimentKey] = _experiments;
+  }
+  return _experiments;
+};
+
+const getAtomType = (nucleus) => nucleus.split(/\d+/)[1];
+
+const getLabel = (correlation) => {
+  let label = Object.keys(lodash.get(correlation, 'attached', {}))
+    .map((otherAtomType) =>
+      correlation.attached[otherAtomType]
+        .map((index) => correlation.label[`${otherAtomType}${index + 1}`])
+        .filter((_label) => _label),
+    )
+    .flat()
+    .filter((_label, i, a) => a.indexOf(_label) === i)
+    .sort((a, b) =>
+      Number(a.split(/[a-z]+/i)[1]) - Number(b.split(/[a-z]+/i)[1]) < 0
+        ? -1
+        : Number(a.split(/[a-z]+/i)[1]) - Number(b.split(/[a-z]+/i)[1]) === 0 &&
+          a.split(/\d+/)[1] < b.split(/\d+/)[1]
+        ? -1
+        : 1,
+    )
+    .join('/');
+
+  if (label.length > 0) {
+    return label;
+  }
+
+  return correlation.label.origin;
+};
+
+const getLabelStyle = (state, correlation) => {
+  if (lodash.get(state, `atomType.${correlation.atomTypes[0]}.error`, false)) {
+    const error = state.atomType[correlation.atomTypes[0]].error;
+    if (
+      lodash
+        .get(error, 'outOfLimit', [])
+        .some((index) => index === correlation.index)
+    ) {
+      return { color: 'red' };
+    }
+    if (
+      lodash
+        .get(error, 'notAttached', [])
+        .some((index) => index === correlation.index)
+    ) {
+      return { color: 'blue' };
+    }
+    if (
+      lodash
+        .get(error, 'ambiguousAttachment', [])
+        .some((index) => index === correlation.index)
+    ) {
+      return { color: 'orange' };
+    }
+  }
+
+  return undefined;
+};
+
+const checkSignalMatch = (signal1, signal2, tolerance) =>
+  signal1.delta - tolerance <= signal2.delta &&
+  signal2.delta <= signal1.delta + tolerance;
+
+const letters = [...Array(26).keys()].map((i) => String.fromCharCode(i + 97));
+
+const attach = (
+  correlations,
+  correlationsAtomType,
+  index,
+  atomTypeToAttach,
+  indicesToAttach,
+) => {
+  const correlation = correlationsAtomType[index];
+  if (!lodash.get(correlation, `attached`, false)) {
+    correlation.attached = {};
+  }
+  if (!lodash.get(correlation, `attached.${atomTypeToAttach}`, false)) {
+    correlation.attached[atomTypeToAttach] = [];
+  }
+  correlation.attached[atomTypeToAttach] = correlation.attached[
+    atomTypeToAttach
+  ]
+    .concat(indicesToAttach)
+    .filter((index, i, a) => a.indexOf(index) === i);
+  if (atomTypeToAttach === 'H') {
+    correlation.attached[atomTypeToAttach].forEach((indexToAttach, n) => {
+      if (lodash.get(correlations, `H[${indexToAttach}]`, false)) {
+        correlations.H[indexToAttach].label[`C${index + 1}`] = `H${index + 1}${
+          letters[n]
+        }`;
+      }
+    });
+  }
```

Comment:
I prefer not to use lodash.get() function if have only one level of keys but it is a very useful tool if you have 2,3 or more keys path